File: MyAngular20.ApplicationService/Views/TurmaView.cs
namespace MyAngular20.ApplicationService.Views
{
    public class TurmaView : ViewBase
    {
        public string Nome { get; set; }
        public int CursoId { get; set; }
        public string Curso { get; set; }
        public short Limite { get; set; }
        public string QtdAtual { get; set; }
        public string Ano { get; set; }
    }

    public interface ITurmaFacade : IViewFacade<TurmaView> { }
}

File: MyAngular20.AppWebAngular/ClientApp/src/app/models/turma-view.ts
export class TurmaView {
  constructor() { }

  id: number = 0;
  nome: string = "";
  ano: string = "";
  cursoId: number = 0;
  curso: string = "";
  limite: number = 0;
  qtdAtual: string = "";
  cadastradoPor: string = "";
  dataCadastro: string = "";
  atualizadoPor: string = "";
  dataAtualizacao: string = "";

  validar(): void {

    let regrasQuebradas: string = "";

    this.atualizadoPor = sessionStorage["nomeUsuario"];

    if (this.nome === "")
      regrasQuebradas += "O campo nome é obrigatório.;";

    if (this.cursoId === 0)
      regrasQuebradas += "O campo curso é obrigatório.;";

    if (this.limite === 0)
      regrasQuebradas += "O campo limite é obrigatório.;";

    if (this.ano === "")
      regrasQuebradas += "O campo ano é obrigatório.;";

    if (this.atualizadoPor === "")
      regrasQuebradas += "O campo atualizado por é obrigatório.;";

    if (regrasQuebradas.length > 0)
      throw new Error(regrasQuebradas);
  }
}
